packages/cli/src/metadata/cache.ts

```diff
@@ -0,0 +1,32 @@
+import { createHash } from 'node:crypto';
+import { readFileSync } from 'node:fs';
+import path from 'node:path';
+import { writeFile, mkdir } from 'node:fs/promises';
+
+const getPath = (repoDir: string, key: string) => `${repoDir}/meta/${key}.json`;
+
+// TODO should we sort keys so that equivalent objects generate the same key?
+const generateKey = (config: any) =>
+  createHash('sha256').update(JSON.stringify(config)).digest('hex');
```

Comment:
Excellent question, would we need it to be recursive as well?
I'm perhaps not that worried as the JSON.parse and the files we generate should stable right? We don't do anything to them along the way...

Thought experiment though, if we stringified it, and sorted all the characters and then hashed it? Would that be stable/equivalent?